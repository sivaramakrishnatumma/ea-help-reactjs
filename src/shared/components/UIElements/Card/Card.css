.title-card {
  display: -webkit-inline-flex;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  min-width: 255px;
  font-size: 16px;
  padding: 8px;
  margin-right: 10px;
  margin-left: 10px;
  margin-bottom: 32px;
  text-align: center;
  vertical-align: top;
  line-height: 20px;
  min-height: 56px;
  height: 40px;
  margin-top: 0;
  position: relative;
  text-decoration: none;
  border-color: #161616;
  border: 2px solid;
  transition: opacity 0.1s;
}

/* .title-card:nth-child(4n) {
  margin-right: 0px !important;
} */

.title-card:hover {
  bottom: -3px;
  left: -3px;
  right: -3px;
  top: -3px;
  width: 261px;
  min-height: 62px;
  margin-right: 4px;
  margin-bottom: 26px;
  cursor: pointer;
}