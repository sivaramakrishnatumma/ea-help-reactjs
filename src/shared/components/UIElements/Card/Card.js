import React from "react";

import "./Card.css";

const Card = props => {
  return (
    <div className={"title-card " + props.className} onClick={props.onClick}>
      <span>{props.title}</span>
    </div>
  );
};

export default Card;
