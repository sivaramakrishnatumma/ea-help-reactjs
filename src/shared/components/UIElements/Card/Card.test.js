import React from "react";
import Card from "./Card";
import { mount } from "enzyme";

describe("Card component test case", () => {
  it("should test card component with default props", () => {
    const props = {
      className: "my-title",
      title: "Sample Title",
      onClick: jest.fn()
    };
    const wrapper = mount(<Card {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
