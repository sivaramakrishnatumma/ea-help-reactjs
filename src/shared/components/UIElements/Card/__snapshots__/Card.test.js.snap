// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test case should test card component with default props 1`] = `
<Card
  className="my-title"
  onClick={[MockFunction]}
  title="Sample Title"
>
  <div
    className="title-card my-title"
    onClick={[MockFunction]}
  >
    <span>
      Sample Title
    </span>
  </div>
</Card>
`;
